src/mod.test.ts

```diff
@@ -0,0 +1,487 @@
+import { test } from 'node:test';
+import assert from 'node:assert';
+import createPlugin, { CallContext, CAPABILITIES } from './mod.ts';
+
+if (typeof WebAssembly === 'undefined') {
+  test('this platform lacks WebAssembly support', async () => {
+    // at the time of writing (2023 Oct 27), playwright webkit builds for windows
+    // do not support webassembly. there's an open PR (https://github.com/WebKit/WebKit/pull/18184)
+    // to fix this though.
+  });
+} else {
+  // The presence of `*.test.ts` files adjacent to module files is no mistake, sadly:
+  // we have to be in the same directory in order to preserve the `__dirname` / `import.meta.url` value
+  // between `mod.ts` and the tests in the build output.
+  test('createPlugin loads a module and provides lookups', async () => {
+    const plugin = await createPlugin('http://localhost:8124/wasm/code.wasm', { useWasi: true });
+
+    try {
+      assert(await plugin.functionExists('count_vowels'), 'count_vowels should exist');
+      assert(await plugin.functionExists(['0', 'count_vowels']), '0:count_vowels should exist');
+      assert(!(await plugin.functionExists(['dne', 'count_vowels'])), 'dne:count_vowels should not exist');
+      assert(!(await plugin.functionExists('count_sheep')), 'count_sheep should not exist');
+    } finally {
+      await plugin.close();
+    }
+  });
+
+  if (!CAPABILITIES.crossOriginChecksEnforced) {
+    test('can create plugin from url with hash check', async () => {
+      const plugin = await createPlugin({
+        wasm: [
+          {
+            url: 'https://raw.githubusercontent.com/extism/extism/v0.5.4/wasm/code.wasm',
+            hash: '7def5bb4aa3843a5daf5d6078f1e8540e5ef10b035a9d9387e9bd5156d2b2565',
+          },
+        ],
+      });
+
+      try {
+        assert.equal(await plugin.functionExists('count_vowels'), true);
+      } finally {
+        await plugin.close();
+      }
+    });
+  }
+
+  test('createPlugin fails on hash mismatch (bad hash)', async () => {
```

Comment:
This test fails because `crypto is not defined`

Reply:
How are you running the tests, out of curiosity?

Reply:
OK, this should be fixed in the latest – Node 18 just needs the `--experimental-global-webcrypto` flag.

Reply:
Ah yeah that's it, I'm on Node 18. I was running them with `npm test`